nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ExtractTNEFAttachments.java

```diff
@@ -0,0 +1,202 @@
+@WritesAttributes({
+        @WritesAttribute(attribute = "filename ", description = "The filename of the attachment"),
+        @WritesAttribute(attribute = "email.tnef.attachment.parent.filename ", description = "The filename of the parent FlowFile"),
+        @WritesAttribute(attribute = "email.tnef.attachment.parent.uuid", description = "The UUID of the original FlowFile.")})
+
+public class ExtractTNEFAttachments extends AbstractProcessor {
+    public static final String ATTACHMENT_ORIGINAL_FILENAME = "email.tnef.attachment.parent.filename";
+    public static final String ATTACHMENT_ORIGINAL_UUID = "email.tnef.attachment.parent.uuid";
+
+    public static final Relationship REL_ATTACHMENTS = new Relationship.Builder()
+            .name("attachments")
+            .description("Each individual attachment will be routed to the attachments relationship")
+            .build();
+    public static final Relationship REL_ORIGINAL = new Relationship.Builder()
+            .name("original")
+            .description("The original file")
+            .build();
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("Flowfiles that could not be parsed")
+            .build();
+    private Set<Relationship> relationships;
+    private List<PropertyDescriptor> descriptors;
+
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_ATTACHMENTS);
+        relationships.add(REL_ORIGINAL);
+        relationships.add(REL_FAILURE);
+        this.relationships = Collections.unmodifiableSet(relationships);
+
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+
+        this.descriptors = Collections.unmodifiableList(descriptors);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+        final ComponentLog logger = getLogger();
+        final FlowFile originalFlowFile = session.get();
+        if (originalFlowFile == null) {
+            return;
+        }
+        final List<FlowFile> attachmentsList = new ArrayList<>();
+        final List<FlowFile> invalidFlowFilesList = new ArrayList<>();
+        final List<FlowFile> originalFlowFilesList = new ArrayList<>();
+
+        session.read(originalFlowFile, new InputStreamCallback() {
+                @Override
+                public void process(final InputStream rawIn) throws IOException {
+                    try (final InputStream in = new BufferedInputStream(rawIn)) {
+                        Properties props = new Properties();
+
+                        HMEFMessage hmefMessage = null;
+
+                            // This will trigger an exception in case content is not a TNEF.
+                            hmefMessage = new HMEFMessage(in);
+
```

Comment:
Not a big deal, but there seem to be formatting/indentation/style issues which makes it a bit difficult to review. Not a big deal since it could be fixed during the merge, but I was wondering if we can help to set you up in such way where you style match the rest of the NiFi. Have you look at this- https://cwiki.apache.org/confluence/display/NIFI/Contributor+Guide#ContributorGuide-CodeStyle. In any event let me know if you have any troubles and we can definitely help.